src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs

```diff
@@ -61,7 +63,8 @@ public class BlockProcessor : IBlockProcessor
             IStorageProvider storageProvider,
             ITxPool txPool,
             IReceiptStorage receiptStorage,
-            ILogManager logManager)
+            ILogManager logManager,
+            IEnumerable<IAdditionalBlockProcessor> additionalBlockProcessors = null)
```

Comment:
maybe Composite pattern CompositeProcessor : IBlockProcessor where it calls on all leaves and can give them scopes when to fire them?

Reply:
Sorry, didn't understand. There is a composite to simplify usage. Used if needed.